### PR TITLE
Rework version selection #18

### LINKED ISSUE #18: Confusion with daily versions
Looking at how the Nextcloud version can be specified, something's a bit off.

The main issue is when choosing the daily channel, namely this part:
`daily: "nextcloud-{{nextcloud_branch}}-daily-{{nextcloud_version}}.zip"`

Looking into the [daily folder](https://download.nextcloud.com/server/daily/) on the download server, this format doesn't exist.

For daily stable releases it should be something like:
`latest-stable{{ nextcloud_version_major }}.zip`

And for bleeding edge daily is should be:
`nextcloud-{{ nextcloud_version_major }}-daily-{{ ansible_date_time.date }}.zip`

I've been working on a PR for this, but I'm not sure how best to tackle it. Main issue is the date for the bleeding edge daily archives, as it's possible that the date is 1 day behind (depending on timezone), resulting in an invalid URL. For that, I was thinking of a fallback system that would go back 1 day in case the archive doesn't exist.
Other option would be to put the date in a variable, allowing to choose an specific daily version, but not sure how necessary that would be. (I guess it could be an optional thing, default being today's date, falling back to yesterday's).

What do you think?

### PR DESCRIPTION
**Pull request by aalaesar**
Fix #18 
- Better variables name
- Default download to latest.zip
- Can technically still download anything available in the nextcloud repo
- Added variable to override the generated download link
- Added variable for changing the file format ( zip, bz2)

### CODE REVIEW

**Comment by aalaesar** [comment]
### TODO
1. Add support for bz2 files
2. Complete readme.

**Comment by aalaesar** [comment]
Hi @noplanman,
I'm about to merge.
Any comment on the reworked variables?

**Comment by noplanman** [comment]
@aalaesar 
Aaargh, sorry, didn't get a chance to check this out yet as I'm abroad at the moment.
Will have a look now!

File: README.md [hunk 1]
```diff
@@ -21,35 +21,74 @@ This role requires root access, so either configure it in your inventory files,
...
-      become: yes
 ```
 
 ## Role Variables
 
 Role's variables (and their default values):
 
-### Installation configuration
-> Source location will be calculated following channel, version and branch values.
+### Choose the version
```
Comment (noplanman) [inline README.md]: This is quite difficult to understand, maybe you could add various examples to elaborate for different scenarios?
Kind of like the Ansible documentation does it.

File: tasks/nc_download.yml [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+- name: "[NC-DL] - bunzip2 is installed"
+  package: name=bzip2 state=installed
+  when: nextcloud_archive_format == "tar.bz2"
+
+- name: "[NC-DL] - regenerate the download URL for *latest*."
+  set_fact:
+    nextcloud_full_url: "{{nextcloud_repository}}/{{nextcloud_version_channel}}/{{[nextcloud_dl_file_name.latest, nextcloud_archive_format]|join('.')}}"
+  when: nextcloud_get_latest and (nextcloud_version_channel != 'prereleases') and (nextcloud_archive_url is not defined)
+
+- name: "[NC-DL] - use the provided URL instead."
```
Comment (noplanman) [inline tasks/nc_download.yml]: I'd move this one before the other one above and then check if `nextcloud_full_url` is set, to skip all the current `when` checks.

Also, why not only use `nextcloud_archive_url` **or** `nextcloud_full_url`? That would save 1 variable name and prevent any ambiguity/confusion between the two.

File: vars/main.yml [hunk 3]
```diff
@@ -1,14 +1,8 @@
...
-  prereleases: "prereleases"
-  daily: "daily"
-  latest: "daily"
-nextcloud_dl_file:
-  releases: "nextcloud-{{nextcloud_version}}.zip"
-  prereleases: "nextcloud-{{nextcloud_version}}.zip"
-  daily: "nextcloud-{{nextcloud_branch}}-daily-{{nextcloud_version}}.zip"
-  latest: "latest-{{nextcloud_branch}}.zip"
-nextcloud_full_url: "{{nextcloud_repository}}/{{nextcloud_dl_path[nextcloud_channel]}}/{{nextcloud_dl_file[nextcloud_channel]}}"
+nextcloud_dl_file_name:
```
Comment (noplanman) [inline vars/main.yml]: I assume you've tested all the different possibilities here? :grin:

File: README.md (hunk 1 above)
Comment (aalaesar) [inline README.md]: Yeah, I also thought about this, this afternoon.
I'll do simple examples just below the vars declarations, then add some variation to the final cases..

File: tasks/nc_download.yml (hunk 2 above)
Comment (aalaesar) [inline tasks/nc_download.yml]: yes, good point!
nextcloud_full_url is defined in the vars folder therefore, always defined.
But maybe i'll simply remove it from vars and put a set_fact task here instead.
I'll use `nextcloud_archive_url`

File: vars/main.yml (hunk 3 above)
Comment (aalaesar) [inline vars/main.yml]: most of them in a dedicated playbook yes. 😄

**Comment by aalaesar** [comment]
Hi,
@noplanman 
I added 2 commits following your reviews.

**Comment by noplanman** [comment]
Nice, great examples in the readme 👍 

Looks good to merge!

**Comment by aalaesar** [comment]
Merged yesterday and released 1.3.0
:+1:  Thank you very much for your contributions so far @noplanman !! :+1:

**Pull request merged**